src/acl/Acl.h

```diff
@@ -42,31 +42,31 @@ void SetKey(SBuf &keyStorage, const char *keyParameterName, const char *newKey);
 /// Can evaluate itself in FilledChecklist context.
 /// Does not change during evaluation.
 /// \ingroup ACLAPI
-class ACL
+class AclNode
```

Comment:
Why is this class renamed to AclNode and not to Acl::Node? Why is it better to modify more than 50 source code files twice (once in this PR and then again in a followup PR that will have to move the same class into its proper namespace) while performing essentially the same renaming change?

Reply:
The reason is searchability. A unique name, although more verbose than needed, enables using many more tools than a clean, but ambiguous without specialist knowledge, name

Reply:
Acl::Node is a globally unique name. Do we really need to argue whether C++ namespaces are better than enhanced searchability by C++-unaware tools?!

Reply:
We don't, and I'm not attached to the name.

<sarcastic, but not wrong>
Thankfully all that's needed now is to `sed -i~ 's/AclNode/Node/g' $(git ls-files)` ; the other way around is not as easy, even with specialised tools
</sarcastic>

If that is the only objection to merging this PR, I'll do that right away.

Reply:
> I'm not attached to the name.

It is not about the spelling of the name. It is about properly using C++ namespaces. FWIW, I was shocked to see AclName being seriously proposed as a candidate when we already have and actively use the Acl namespace.

> Thankfully all that's needed now is to `sed -i~ 's/AclNode/Node/g' $(git ls-files)`

That substitution would not work AFAICT because AclNode was not declared inside the Acl namespace. You need something like `s/AclNode/Acl::Node/g`.

> the other way around is not as easy, even with specialised tools

Mechanical[^1] renaming of official ACL to Acl::Node is as easy/difficult as mechanical renaming of official ACL to AclNode. The primary difference is that only one of the two targets is acceptable for long-term use.

[^1]: Here, I use that term to mean "without the use of tools that understand C++".

Mechanical renaming of Acl::Node to Acl::Foo is currently also easy because there are very few bare Node names in the current code.

In general, yes, namespaces may complicate or prohibit certain mechanical actions. That is a (small) negative to consider when evaluating whether to introduce a _new_ namespace. However, those difficulties are pretty much irrelevant when it comes to using the _existing_ applicable namespace!

> If that is the only objection to merging this PR, I'll do that right away.

I reserve the right to request (or implement) more changes after I see (or implement) the outcome of this critical correction. The compiler might ask us to fix a few problems as well. However, FWIW, I will not insist on you removing unnecessary `Acl::` prefixes from Acl::Node occurrences inside Acl namespace; that "Foo inside Foo" style violation can be considered acceptable for this special renaming PR IMO.

Reply:
AclNode -> Acl::Node is now done, including hoisting the class into its own header. Unfortunately the implementation of the ACL types registry is too coupled to Node's internals to do the same with the implementation file without out-of-scope surgery